### PR TITLE
Implement XOAUTH2 authentication method

### LINKED ISSUE #31: Support XOAUTH2
Google [encourages](https://developers.google.com/gmail/oauth_overview) the use of the login mechanism XOAUTH2 [1]. It would be great if Net_SMTP had support for this.

Some informations about the protocol and SMTP can be found here: https://developers.google.com/gmail/xoauth2_protocol#smtp_protocol_exchange

### PR DESCRIPTION
**Pull request by thomascube**
Takes an OAuth access token as the passwort argument
to authenticate using the XOAUTH2 mechanism as
suggested by Gmail and Office 365.

Fixes #31

So far tested with Gmail, now submitting for review.

### CODE REVIEW

**Comment by jparise** [comment]
I just fixed up the Travis CI config. @thomascube, please rebase, and you should get a green build.

**Comment by thomascube** [comment]
@jparise Done and tests seem to pass now. Looking forward to your review or merge. Thanks!

File: Net/SMTP.php [hunk 1]
```diff
@@ -965,6 +966,45 @@ protected function authGSSAPI($uid, $pwd, $authz = '')
...
+    /**
+     * Authenticates the user using the XOAUTH2 method.
+     *
+     * @param string $uid   The userid to authenticate as.
+     * @param string $token The access token to authenticate with.
+     * @param string $authz The optional authorization proxy identifier.
+     *
+     * @return mixed Returns a PEAR_Error with an error message on any
+     *               kind of failure, or true on success.
+     * @since 1.8.2
```
Comment (jparise) [inline Net/SMTP.php]: I might tag the next release as 1.9.0 to reflect this new feature, but I can update this as part of the release process.

Comment (thomascube) [reply Net/SMTP.php]: Yeah, that was just an assumption I made. Is there an ETA for the next release?

Comment (jparise) [reply Net/SMTP.php]: @thomascube I just released version 1.9.0.

**Pull request merged**